src/ccl/charts/line-graph.component.ts

```diff
@@ -0,0 +1,119 @@
+import { Component, ViewEncapsulation, ElementRef } from '@angular/core';
+import * as D3 from 'd3';
+
+/*
+ * Line graph component
+ * Contains all logic for drawing a line graph
+ */
+
+@Component({
+  selector: 'line-graph',
+  encapsulation: ViewEncapsulation.None,
+  template: `<ng-content></ng-content>`,
+  inputs: [ 'data', 'indicator' ],
+  styleUrls: [ '../../assets/css/chart.css']
+})
+
+export class LineGraph {
+
+  public data: Object;
```

Comment:
It would be nice to take advantage of typescript here and give this a type, something like `ChartData`? Or `LineGraphData`? This way its more clear to the developer what the input for this component is, and typescript can validate it at compile time.